### DISCUSSION TITLE
Support renaming of "sources" shown in the Energy dashboard

### DISCUSSION THREAD

**Discussion by andreas-bulling** [home-assistant/frontend#18215] [Dashboard] [answered]
It would be great if the name of sources selected to be shown/used in the Energy dashboard could be "overwritten" in the form of a "display name". In particular those devices monitored individually.

At the moment the entity name is used.

In my case these names follow a certain scheme to be able to easily tell them apart/sort them in a large installation. But these names are not suited/nice to see on the dashboard. So it'd be great to have the option to provide a "display name" or something in the dashboard settings, when adding a source.

**Comment by karwosts** [comment]
Custom names for individual energy devices merged for 2024.5 ([#20033](https://github.com/home-assistant/frontend/pull/20033))

**Reply by chocomega** [reply]
@karwosts I have not yet installed 2024.5 but does this feature also apply to energy sources ? It seems only available for individual devices.

**Reply by andreas-bulling** [reply]
I was just about to write exactly the same - it'd be great if this functionality was also available for energy sources. They tend to have cryptic names.

**Reply by karwosts** [reply]
I haven't added to things other than individual devices. Those were kind of the most obvious to start with given how prominent the names are. May consider expanding it to others in the future.

**Reply by chocomega** [reply]
Alright, thanks for the clarification. It would be great indeed to expand that to others as well, especially for some sources which can not be renamed at all e.g. sources not entity based but from external statistics with the form `stat_domain:stat_id`

**Reply by andreas-bulling** [reply]
Adding support for energy sources would be fantastic @karwosts

**Reply by spitsy1** [reply]
Any chance this will be possible in future update? Because now on smaet phone i need to scroll to the page to see the figures,  as you see below.
![Screenshot_20250202_101513_Home Assistant](https://github.com/user-attachments/assets/feb42e09-ceff-490a-8eb6-a301fa7c4594)
Would be great!

**Reply by Sven177** [reply]
Try to click on that "envoy 1221xxxxx" - and then in the pop up dialog click at the gearwheel-icon. Then there should be an input field for a name to be displayed (instead of the entity name)...

**Comment by karwosts** [comment, marked as answer]
All energy/gas/water/solar/battery sources will be custom nameable for energy in 2026.6.
